### PR TITLE
Introduces reset-kibana-system-user tool

### PR DESCRIPTION
**Pull request by jkakavas**
This commit adds a new CLI tool `elasticsearch-reset-kibana-system
-tool` that can be used to set and reset the password of the
kibana-system built-in user.

### CODE REVIEW

File: x-pack/plugin/security/src/main/bin/elasticsearch-reset-kibana-system-password [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+ES_MAIN_CLASS=org.elasticsearch.xpack.security.authc.esnative.tool.ResetBuiltinPasswordTool \
```
Comment (jkakavas) [inline x-pack/plugin/security/src/main/bin/elasticsearch-reset-kibana-system-password:8]: An additional CLI tool is what made most sense after our discussion today. In the long run we can re-consider the possibility of a single tool for all builtin users ( replacing setup-passwords ) but for now I presume this has small added value.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetBuiltinPasswordTool.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 import joptsimple.OptionSet;
 
+import joptsimple.OptionSpec;
```
Comment (jkakavas) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetBuiltinPasswordTool.java:12]: Generifying the existing class made more sense than duplicating it.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetBuiltinPasswordTool.java [hunk 3]
```diff
@@ -34,100 +35,107 @@
 import static org.elasticsearch.xpack.security.tool.CommandLineHttpClient.createURL;
 import static org.elasticsearch.xpack.security.tool.CommandUtils.generatePassword;
 
-public class ResetElasticPasswordTool extends BaseRunAsSuperuserCommand {
+public class ResetBuiltinPasswordTool extends BaseRunAsSuperuserCommand {
```
Comment (jkakavas) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetBuiltinPasswordTool.java:38]: Generifying the existing class made more sense than duplicating it.

File: x-pack/plugin/security/src/main/bin/elasticsearch-reset-kibana-system-password (hunk 1 above)
Comment (tvernum) [reply x-pack/plugin/security/src/main/bin/elasticsearch-reset-kibana-system-password:8]: That was my feeling as well - we should have a conversation about a generic "reset-user-password" tool. 
But there are questions:
- is it "reset many users at once" ?
- is it only reserved users, or native users as well?
- does it have a batch mode for scripting?

And today is not the day to decide on those.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetBuiltinPasswordTool.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+    public static void main(String[] args) throws Exception {
+        exit(new ResetBuiltinPasswordTool().main(args, Terminal.DEFAULT));
+    }
+
+    protected ResetBuiltinPasswordTool(
+        Function<Environment, CommandLineHttpClient> clientFunction,
+        CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction
+    ) {
+        super(clientFunction, keyStoreFunction, "Resets the password of the elastic built-in user");
```
Comment (tvernum) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetBuiltinPasswordTool.java:56]:
```suggestion
        super(clientFunction, keyStoreFunction, "Resets the password of a built-in user");
```

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetKibanaSystemPasswordTool.java [hunk 5]
```diff
@@ -34,25 +33,25 @@
 import static org.elasticsearch.xpack.security.tool.CommandLineHttpClient.createURL;
 import static org.elasticsearch.xpack.security.tool.CommandUtils.generatePassword;
 
-public class ResetElasticPasswordTool extends BaseRunAsSuperuserCommand {
+public class ResetKibanaSystemPasswordTool extends BaseRunAsSuperuserCommand {
```
Comment (tvernum) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetKibanaSystemPasswordTool.java:36]: Should this file exist? I assume not.

**Review by tvernum — APPROVED** [review]
LGTM

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetKibanaSystemPasswordTool.java (hunk 5 above)
Comment (jkakavas) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/tool/ResetKibanaSystemPasswordTool.java:36]: agressive ctrl+z ? Not sure :shrug:

**Pull request merged**